pkg/net/net.go

```diff
@@ -224,6 +280,26 @@ func (o *Overlay) handleSample(s network.Stream) {
 	o.n.Handle(o.ctx(), o.log.WithStream(s), o.h, s)
 }
 
+func (o *Overlay) handleGossip(s network.Stream) {
+	defer s.Close()
+
+	g := gossip{
+		o.n, o.h, o.h, s.Conn().RemotePeer(), protoutil.Join(o.proto, "gossip"),
+	}
+	peers, err := g.Pull(o.ctx(), s)
```

Comment:
Can we do `g.Pull` and `g.Push` concurrently, _e.g._ using [`errgroup.Group`](https://pkg.go.dev/golang.org/x/sync/errgroup)?  It seems like `Pull` is just reading and `Push` is just writing.

Reply:
Good point! Yes, we can, and we will.